### ISSUE TITLE
[Bug]: Invalid content doesn't cause error

### ISSUE THREAD

**Issue by lepult** [ueberdosis/tiptap#5500]
### Affected Packages

core, paragraph

### Version(s)

2.6.3

### Bug Description

Content that doesn't follow the schema can be set with no errors. 

### Browser Used

Firefox

### Code Example URL

https://codesandbox.io/p/sandbox/friendly-leaf-ksgwwk

### Expected Behavior

In the sandbox the Tiptap content is set to a Paragraph that contains a HorizontalRule and another Paragraph. This content is invalid, since the schema of Paragraph is 'inline*', while both Paragraph and HorizontalRule are 'block' nodes.

Despite that, the content is set without error and the invalid content is displayed. After you type something and redo that change (CTRL+Z) an error is thrown (when you write in the first line, the error is thrown instantly). Why is that error only thrown after changes and not when the content is set?
![grafik](https://github.com/user-attachments/assets/8bfa75a3-a22a-4bba-9ccb-65e755989cdc)

### Additional Context (Optional)

This is the result of editor.getJSON() after initialisation of the editor. You can see how the content is invalid, since there is a Paragraph and a HorizontalRule within the Paragraph.
```
{
  "type": "doc",
  "content": [
    {
      "type": "paragraph",
      "content": [
        {
          "type": "text",
          "text": "Before horizontalRule"
        },
        {
          "type": "horizontalRule"
        },
        {
          "type": "text",
          "text": "After horizontalRule"
        },
        {
          "type": "paragraph",
          "content": [
            {
              "type": "text",
              "text": "Text in Paragraph"
            }
          ]
        }
      ]
    }
  ]
}
```

### Dependency Updates

- [X] Yes, I've updated all my dependencies.

**Comment by nperez0111** [comment]
So what `contentCheck` is checking for is invalid nodes & marks that are not defined in your schema, not that they follow your schema's nesting rules.

To check for that, it is a pretty easy change:
```diff
diff --git a/packages/core/src/helpers/createNodeFromContent.ts b/packages/core/src/helpers/createNodeFromContent.ts
index 7fe83f83c..63a9a2f9d 100644
--- a/packages/core/src/helpers/createNodeFromContent.ts
+++ b/packages/core/src/helpers/createNodeFromContent.ts
@@ -45,7 +45,13 @@ export function createNodeFromContent(
         return Fragment.fromArray(content.map(item => schema.nodeFromJSON(item)))
       }
 
-      return schema.nodeFromJSON(content)
+      const node = schema.nodeFromJSON(content)
+
+      if (options.errorOnInvalidContent) {
+        node.check()
+      }
+
+      return node
     } catch (error) {
       if (options.errorOnInvalidContent) {
         throw new Error('[tiptap error]: Invalid JSON content', { cause: error as Error })
```
But this will still show the invalid content, ideally prosemirror would just skip over any content that it considers invalid and allows you to display the rest. But it only throws an error right now saying that the node nesting is invalid.

This will require some more thought

**Comment by lepult** [comment]
Thank you for your answer. I didn't know about the node.check() function. My use case is only concerned with detecting invalid content and not with its display, so your answer is a great help to me and already resolves my issues.

**Comment by nperez0111** [comment]
If that is the case, I would be happy to accept a PR with this change, it requires minimal work on our side and I think matches user expectation with `enableContentCheck` enabled

**Comment by lepult** [comment]
Do you want me to submit that PR?

**Comment by nperez0111** [comment]
That would be great @lepult

**Comment by nperez0111** [comment]
This has been released with 2.7.0-pre.0 and will be released in tiptap 2.7.0 when this beta has been tested by more users
